assets/js/base/utils/errors.js

```diff
@@ -1,9 +1,22 @@
-export const formatError = ( apiError ) => {
-	return typeof apiError === 'object' && apiError.hasOwnProperty( 'message' ) ? {
-		apiMessage: apiError.message,
-	} : {
-		// If we can't get any message from the API, set it to null and
-		// let <ApiErrorPlaceholder /> handle the message to display.
-		apiMessage: null,
-	};
+const formatErrorMessage = ( error, messageProperty = 'frontendMessage' ) => {
```

Comment:
I think it'd be useful to have a doc block here, especially for describing what `messageProperty` is a reference to and acceptable values.

Also I'd be wary of getting to context specific with `frontendMessage` string here.  What about `userFriendly` instead (I'm guessing the idea here is a non-dev type error message?)

Reply:
`frontendMessage` was referring to errors produced by the frontend, but not necessarily user-friendly. For example, a JS error.

I completely refactored that part of the code in 2ac40b02d47db9e640d3eee79b95aa4b115f482a, let me know how it looks now.

Reply:
> frontendMessage was referring to errors produced by the frontend, but not necessarily user-friendly. For example, a JS error

That's a good argument for _not_ using `frontend` though as it's still ambiguous.  In continuing with the intended purpose, I would expect it to be something like `general` (for general JS error) as opposed to `api` (for error in an api response).  

I have some more feedback on the refactor (including a suggestion to continue iterating on it in a separate pull so we don't block this one).